src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -181,10 +181,18 @@ private bool IsCompatibleWithTarget(NuGetFramework target, NuGetFramework candid
 
         private static bool IsCompatibleWithTargetCore(NuGetFramework target, NuGetFramework candidate)
         {
-            // compare the frameworks
-            return (NuGetFramework.FrameworkNameComparer.Equals(target, candidate)
-                && StringComparer.OrdinalIgnoreCase.Equals(target.Profile, candidate.Profile)
-                && IsVersionCompatible(target.Version, candidate.Version));
+            if (target.IsNet5Era)
+            {
+                return NuGetFramework.FrameworkNameComparer.Equals(target, candidate)
+                    && IsVersionCompatible(target.Version, candidate.Version)
+                    && (!candidate.HasProfile || StringComparer.OrdinalIgnoreCase.Equals(target.Profile, candidate.Profile));
+            }
+            else
+            {
+                return NuGetFramework.FrameworkNameComparer.Equals(target, candidate)
+                    && IsVersionCompatible(target.Version, candidate.Version)
+                    && StringComparer.OrdinalIgnoreCase.Equals(target.Profile, candidate.Profile);
+            }
```

Comment:
I understand that this code is developed for better readability. I prefer below logic  because in the future we have to maintain almost same logic in `if..else` block that can lead to maintainability problems. Thoughts?
```cs
bool result = NuGetFramework.FrameworkNameComparer.Equals(target, candidate) &&
                      IsVersionCompatible(target.Version, candidate.Version);

result = result && ((target.IsNet5Era && !candidate.HasProfile) || 
                     StringComparer.OrdinalIgnoreCase.Equals(target.Profile, candidate.Profile));

return result;

```

Reply:
will consider in 2nd wave. trying to merge to make the train.

Reply:
will consider in 2nd wave via this follow up issue: https://github.com/nuget/home/issues/9260